src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/ConstructorBuilder.cs

```diff
@@ -27,7 +27,7 @@ public sealed class ConstructorBuilder : ConstructorInfo
         }
 
         internal ConstructorBuilder(string name, MethodAttributes attributes, CallingConventions callingConvention,
-            Type[]? parameterTypes, ModuleBuilder mod, TypeBuilder type) :
+            Type[]? parameterTypes, ModuleBuilder mod, [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] TypeBuilder type) :
```

Comment:
Why All? Shouldn't this be actually nothing

Reply:
Reflection.Emit is annotated as requiring all members everywhere - it allows us to also have returns that provide all members (e.g. `private TypeBuilder GetTypeBuilder()` is allowed to return `DynamicallyAccessedMemberTypes.All` on line 46 below). This eliminates the need for suppressions. It's pretty easy to ensure all TypeBuilders we have in the system are annotated as `.All` because that's how they're "born".

Ref.Emit cannot really be used safely unless the entire type is available for all cases that matter (base types, interfaces, etc.).

An "unnecessary" annotation on an internal API feels better than a suppression that requires one to think "is this really safe to suppress now?" and "can it become not safe to suppress in the future?".